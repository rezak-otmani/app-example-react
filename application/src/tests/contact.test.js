import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Contact from '../views/contact/contact';


describe('Contact Component', () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Contact />
    );
  });


            it('it should render the title', () => {
             const item = wrapper.find('h1').first();
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('GET IN TOUCH')
           });


           it('it should render Contact section ', () => {
              expect(wrapper.length).toBe(1);

          });

             
           
 describe('Contact information section', () => {


           it('shows item mail ', () => {
             const item = wrapper.find('.item').at(0);
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('info@mysite.com ')
           });
           
            it('shows item tel', () => {
             const item = wrapper.find('.item').at(2);
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('Tel: 123-456-7890')
           });

            it('shows item adresse ', () => {
             const item = wrapper.find('.item').at(4);
             expect(item.length).toEqual(1);
             expect(item.childAt(0).text()).toBe('500 Terry Francois Street')
           });

             

});


describe('Map section', () => {
         
      it('it should render the map component', () => {
             const map = wrapper.find('MapContainer').first();
             expect(map.length).toEqual(1);
           });
});


           const tree = renderer.create(
                      <Contact />
                          ).toJSON();
                  expect(tree).toMatchSnapshot();


});
