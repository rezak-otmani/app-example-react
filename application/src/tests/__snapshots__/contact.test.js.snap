// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="pusher"
>
  <div
    className="ui vertical olive inverted masthead center aligned segment"
  >
    <div
      className="center aligned column"
    >
      <div
        className="ui sizer vertical segment"
      >
        <h1
          className="ui huge header"
        >
          GET IN TOUCH
        </h1>
      </div>
    </div>
    <div
      className="ui seven columns center aligned stackable divided equal height stackable grid"
    >
      <div
        className="three wide column"
      >
        <div
          className="ui link list"
        >
          <a
            className="item"
            href=""
          >
            info@mysite.com 
          </a>
          <a
            className="item"
            href=""
          >
            info@mysite.com
          </a>
        </div>
      </div>
      <div
        className="three wide column"
      >
        <div
          className="ui link list"
        >
          <a
            className="item"
            href=""
          >
            Tel: 123-456-7890
          </a>
          <a
            className="item"
            href=""
          >
            Fax: 123-456-7890
          </a>
        </div>
      </div>
      <div
        className="three wide column"
      >
        <div
          className="ui link list"
        >
          <p
            className="item"
          >
            500 Terry Francois Street
          </p>
          <p
            className="item"
          >
            San Francisco, CA 94158
          </p>
        </div>
      </div>
    </div>
    <div
      className="ui billboard test ad"
      data-text=""
    >
      <div>
        <div
          className={undefined}
          style={
            Object {
              "height": "100%",
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <div
            style={
              Object {
                "bottom": 0,
                "display": "inherit",
                "height": "100vh",
                "left": 0,
                "position": "relative",
                "right": 0,
                "top": 0,
                "width": "100vw",
              }
            }
          >
            Loading map...
          </div>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
